src/image/loading_displaying.js

```diff
@@ -158,6 +159,341 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
   return pImg;
 };
 
+/**
+ * Generates a gif of your current animation and downloads it to your computer!
+ *
+ * The duration argument specifies how many seconds you want to record from your animation.
+ * This value is then converted to the necessary number of frames to generate it.
+ *
+ * With the delay argument, you can tell the function to skip the first `delay` seconds
+ * of the animation, and then download the `duration` next seconds. This means that regardless
+ * of the value of `delay`, your gif will always be `duration` seconds long.
+ *
+ * @method saveGif
+ * @param  {String} filename File name of your gif
+ * @param  {Number} duration Duration in seconds that you wish to capture from your sketch
+ * @param  {Number} delay Duration in seconds that you wish to wait before starting to capture
+ *
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(100, 100);
+ *   colorMode(HSL);
+ * }
+
+ * function draw() {
+ *   // create some cool dynamic background
+ *   let hue = map(sin(frameCount / 100), -1, 1, 0, 100);
+ *   background(hue, 40, 60);
+
+ *   // create a circle that moves diagonally
+ *   circle(
+ *     100 * sin(frameCount / 10) + width / 2,
+ *     100 * sin(frameCount / 10) + height / 2,
+ *     10
+ *   );
+ * }
+
+ * // you can put it in the mousePressed function,
+ * // or keyPressed for example
+ * function mousePressed() {
+ *   // this will download the first two seconds of my animation!
+ *   saveGif('mySketch', 2);
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * animation of a circle moving smoothly diagonally
+ */
+p5.prototype.saveGif = async function(fileName, seconds = 3, delay = 0) {
```

Comment:
We probably shouldn't force it to be an integer in case, for example, we have a perfect loop in 90 frames, or 1.5s. (Although I think forcing it to be an integer would make sense if we use frames as a unit instead of seconds, which could also be an option)

Reply:
Using frames as units is incredibly useful for advanced users and not intuitive for newcomers. Having the ability to switch between the two seems ideal. How do we do that? Using another argument like `frameUnits = true`?

Reply:
Maybe adding an optional object with extra options at the end? e.g.:
```js
function(fileName, time, { delay = 0, units = 'seconds' }) {
  const frames = units === 'seconds' ? time * 60 : time
  // etc
}

// Usage:
saveGif('test.gif', 30, { units: 'frames' })
```

Might make it easier to read at call sites with the key: value syntax rather than having `true` as a parameter without extra context around it. P5 does something similar when there are advanced options, e.g. in `textToPoints`: https://p5js.org/reference/#/p5.Font/textToPoints

Reply:
Sounds ideal!

Reply:
yes definetly makes sense. @jesi-rgb we can do that !